clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,449 @@
+}
+
+func (*networkPerformanceMeasurement) String() string {
+	return networkPerformanceMetricsName
+}
+
+func (npm *networkPerformanceMeasurement) storeWorkerPods() error {
+	pods, err := npm.k8sClient.CoreV1().Pods(netperfNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		if pod.Status.PodIP == "" {
+			klog.Errorf("IP address not found for %s", pod.Name)
+			continue
+		}
+		podData := &workerPodData{podName: pod.Name, podIP: pod.Status.PodIP, workerNode: pod.Spec.NodeName}
+		npm.populateWorkerPodList(podData)
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) gather() (measurement.Summary, error) {
+	close(npm.stopCh)
+	npm.cleanupCluster()
+	npm.createResultSummary()
+	content, err := util.PrettyPrintJSON(&measurementutil.PerfData{
+		Version:   "v1",
+		DataItems: npm.resultSummary.DataItems,
+	})
+	if err != nil {
+		klog.Info("Pretty Print to Json Err: ", err)
+	}
+	summaryName := fmt.Sprintf(npm.String() + npm.resultSummary.podRatio + npm.resultSummary.protocol + npm.resultSummary.service)
+	return measurement.CreateSummary(summaryName, "json", content), nil
+}
+
+// Dispose disposes resources,objects after the measurement.
+func (npm *networkPerformanceMeasurement) Dispose() {
+	if npm.framework == nil {
+		klog.V(1).Infof("Network measurement %s wasn't started, skipping the Dispose() step", npm)
+		return
+	}
+	klog.Infof("Stopping network performance measurement...")
+	npm.cleanupCluster()
+}
+
+func (npm *networkPerformanceMeasurement) handleUpdateNetworkTestEvents(newObj interface{}) {
+	newRuntimeObj, ok := newObj.(runtime.Object)
+	if newObj != nil && !ok {
+		klog.Errorf("error casting object: %s", newObj)
+		return
+	}
+	resourceContent, err := runtime.DefaultUnstructuredConverter.ToUnstructured(newRuntimeObj)
+	if err != nil {
+		klog.Errorf("error converting event Object to map representation: %s", newObj)
+		return
+	}
+	var metricResp metricResponse
+	metricMap, ok := resourceContent["status"].(map[string]interface{})
+	if !ok {
+		klog.Errorf("status Object not found")
+		return
+	}
+	if err = getMetricResponse(metricMap, &metricResp); err != nil {
+		klog.Error(err)
+	}
+	if len(metricResp.Metrics) == 0 && metricResp.Error != "" {
+		klog.Error("error message from worker pod: ", metricResp.Error)
+	}
+	npm.metricLock.Lock()
+	defer npm.metricLock.Unlock()
+	npm.metricVal = append(npm.metricVal, metricResp.Metrics)
+}
+
+func (npm *networkPerformanceMeasurement) execNToMTest() {
+	npm.podInfo.formUniquePodPair()
+	npm.createCustomResourcePerUniquePodPair()
+}
+
+func (npm *networkPerformanceMeasurement) createCustomResourcePerUniquePodPair() {
+	currTime := time.Now()
+	npm.startTimeStampForTestExecution = currTime.Add(initialDelayForTestExecution).Unix()
+	for pairIndex, podPair := range npm.podInfo.uniquePodPairList {
+		npm.templateMapping = npm.populateTemplate(podPair, pairIndex)
```

Comment:
you shouldn't store tamplteMapping to the npm field here; just do:
```
templateMapping := np.populateTemplate(...)
if err := ...
```

Reply:
modified